specification/common-types/resource-management/v4/types.json

```diff
@@ -0,0 +1,666 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "3.0",
+    "title": "Common types"
+  },
+  "paths": {},
+  "definitions": {
+    "Resource": {
+      "title": "Resource",
+      "description": "Common fields that are returned in the response for all Azure Resource Manager resources",
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
Should we take advantage of making new v4 types to use the [new arm-id format type](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-arm-id-details)?

Reply:
Added `format: arm-id` but didn't add `x-ms-arm-id-details` as it probably would be something like this:

```
"x-ms-arm-id-details": {
  "allowedResources": [
    {
       "scopes": [],
      "type": "*"
    }
  ]
}
```

Reply:
yeah that makes sense